packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -717,11 +731,17 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
   }
 }
 
+typedef bool SelectableDayPredicate(DateTime day);
```

Comment:
We need docs here.  Doesn't need to be long.  Usually we say something like "Signature for XX.\n\nUsed by YYY."  That way folks who see this on the web site can click through to showDatePicker and be on the right track.